packages/api/src/objectSet/ObjectSet.test.ts

```diff
@@ -78,6 +220,15 @@ describe("ObjectSet", () => {
         .toEqualTypeOf<number>();
     });
 
+    it("can be sub-selected", () => {
+      const objectWithUndefinedRdp = fauxObjectSet.withProperties({
+        "derivedPropertyName": (base) =>
+          base.pivotTo("lead").selectProperty("employeeId"),
+      }).fetchOne(3, {
+        $select: ["derivedPropertyName"],
+      });
+    });
```

Comment:
A version of this test broke for me in @osdk/client so I wrote one in @osdk/api to iterate faster.